### PR TITLE
Fix name collisions in objects from libs/

### LINKED ISSUE #71: [BUG] Using libs/ with the same source file name quietly fails (possibly with a link error if uses differ)
**Describe the bug**

When using the libs/ source layout, repeating the same filename between multiple submodules causes dds to quietly compile overwrite the .o file for one of the files.

From what I can tell, the [pitchfork layout does not require](https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs#submod.libs) that the submodules don't emit conflicting symbols / have conflicting filenames.

As such, the expected behavior is to either successfully compile (namespacing by each libs/ submodule), or at minimum emit an error explaining the issue.

**Steps to Reproduce**

This zipfile contains the project setup, which is also replicated following: [repro-dds-multilib-dep.zip](https://github.com/vector-of-bool/dds/files/6218302/repro-dds-multilib-dep.zip)

Directory structure:
```
libs/
  hello/
    src/
      hello.cpp
      hello.test.cpp
    library.json5
  hello2/
    src/
      hello.cpp
      hello2.test.cpp
    library.json5
package.json5
```

Of note, if `hello2.test.cpp` is named `hello.test.cpp`, only one test executable will be generated, which is also a case which I feel matches this bug report. For the sake of simplicity, it is omitted.

package.json5
```json5
{
    name: 'some-project',
    namespace: 'example',
    version: '0.0.0',
    depends: ['spdlog@1.4.0'],
}
```

libs/hello/library.json5
```json5
{
    name: 'hello',
    uses: ['spdlog/spdlog']
}
```

libs/hello/src/hello.cpp
```c++
#include <string>

#include <spdlog/spdlog.h>

namespace hello
{
    std::string message()
    {
        spdlog::info("hello::message() called");
        return "Hello, World!";
    }
}
```

libs/hello/src/hello.test.cpp
```c++
#include <string>

namespace hello
{
    std::string message();
}

int main()
{
    hello::message();
}
```

libs/hello2/library.json5
```json5
{
    name: 'hello2'
}
```

libs/hello2/src/hello.cpp
```c++
#include <string>

namespace hello
{
    std::string message()
    {
        return "Hello, World!";
    }
}
```

libs/hello2/src/hello2.test.cpp
```
#include <string>

namespace hello
{
    std::string message();
}

int main()
{
    hello::message();
}
```

Compile the project ( `dds build -t:debug:gcc` ), and observe the link failure message when attempting to link test/hello2

**`dds` Output**

```
[warning] /home/jbassett/.local/share/dds/pkg/vob-semester@0.2.2/package.jsonc: Using a JSON object for 'depends' is deprecated. Use an array of strings instead.
[warning] /home/jbassett/.local/share/dds/pkg/vob-semester@0.2.2/package.jsonc: Using a JSON object for 'depends' is deprecated. Use an array of strings instead.
[info ] NOTE: A prior version of the project build database was found.
[info ] This is not an error, but incremental builds will be invalidated.
[info ] The database is being upgraded, and no further action is necessary.
[info ] [example/hello2] Compile: hello.cpp
[info ] [fmt/fmt] Compile: posix.cc
[info ] [fmt/fmt] Compile: format.cc
[info ] [example/hello] Compile: hello.cpp
[info ] [example/hello2] Compile: hello2.test.cpp
[info ] [example/hello] Compile: hello.test.cpp
[info ] [example/hello] Compile: hello.test.cpp             -     296ms [1/6]
[info ] [fmt/fmt] Compile: format.cc                        -    6009ms [2/6]
[info ] [example/hello2] Compile: hello2.test.cpp           -    6009ms [3/6]
[info ] [example/hello] Compile: hello.cpp                  -    6009ms [5/6]
[info ] [fmt/fmt] Compile: posix.cc                         -    6009ms [4/6]
[info ] [example/hello2] Compile: hello.cpp                 -    6009ms [6/6]
[info ] Compilation completed in 6075ms
[info ] [example/hello] Archive: libhello.a
[info ] [example/hello2] Archive: libhello2.a
[info ] [fmt/fmt] Archive: _deps/fmt@6.0.0/libfmt.a
[info ] [example/hello] Archive: libhello.a - 55ms
[info ] [example/hello2] Archive: libhello2.a - 55ms
[info ] [fmt/fmt] Archive: _deps/fmt@6.0.0/libfmt.a - 55ms
[info ] Archiving completed in 56ms
[info ] [example/hello] Link: test/hello                    
[info ] [example/hello2] Link: test/hello2                   
[info ] [example/hello] Link: test/hello                     -    342ms
[info ] [example/hello2] Link: test/hello2                    -    342ms
[error] argument not found
[error] Linking a runtime binary (executable/shared library/DLL) failed
[error] 
Linking a runtime binary file failed. There are a variety of possible causes
for this error. Refer to the documentation for more information.

[error] Refer: https://vector-of-bool.github.io/docs/dds/err/link-failure.html
```

It is fairly clear that the missing runtime binary is spdlog. E.g. adding `uses: ['spdlog/spdlog']` to hello2/library.json5 successfully links, but test/hello2 prints out the log message in hello/src/hello.cpp.

**Desktop (please complete the following information):**

 - Operating System: Ubuntu 20.04
 - Compiler: GCC
 - `dds` Version: `0.1.0-alpha.6`

**Additional context**

dds is overwriting hello2/src/hello.cpp's object file when compiling hello/src/hello.cpp.

### PR DESCRIPTION
**Pull request by Quincunx271**
Fixes #71 

Apparently I got most of the way implementing this a while ago. Solves the issue by including the path from root in the output files of library roots under libs/. In a non-libs/ layout, the output paths should be unaffected.

### CODE REVIEW

**Comment by vector-of-bool** [comment]
It looks like the Conan endpoint that was being used to download Windows OpenSSL binaries has moved. Could you update that URL in this branch? The old dead URL is in the `tools/get-win-openssl.ps1` file. The new URL should be `https://center.conan.io/artifactory/api/conan/conan-center/v1/files/_/openssl/1.1.1h/_/986a09a7b68304213376088bd560b672/package/a79a557254fabcb77849dd623fed97c9c5ab7651/33ddaecbc11a8cef4eca0c21a9827f99/conan_package.tgz`

**Comment by Quincunx271** [comment]
> It looks like the Conan endpoint that was being used to download Windows OpenSSL binaries has moved. Could you update that URL in this branch?

Done, but it still seems to error out. The error message changed, though; it now says `403 Forbidden`. I tested `wget "$openssl_url"` on my linux machine (after translating the two variables to bash), and it worked.

**Comment by Quincunx271** [comment]
Anything that still needs to be done for this PR? I'd like to get it merged, but unfortunately I don't have a Windows machine to debug this on.

File: tools/get-win-openssl.ps1 [hunk 1]
```diff
@@ -12,8 +12,8 @@ New-Item -ItemType Container $build_dir -ErrorAction Ignore
 $local_tgz = Join-Path $build_dir "openssl.tgz"
 
 # This is the path to the release static vs2019 x64 build of OpenSSL in bintray
-$conan_ssl_path = "_/openssl/1.1.1h/_/7098aea4e4f2247cc9b5dcaaa1ebddbe/package/a79a557254fabcb77849dd623fed97c9c5ab7651/141ef2c6711a254707ba1f7f4fd07ad4"
-$openssl_url = "https://dl.bintray.com/conan/conan-center/$conan_ssl_path/conan_package.tgz"
+$conan_ssl_path = "_/openssl/1.1.1h/_/986a09a7b68304213376088bd560b672/package/a79a557254fabcb77849dd623fed97c9c5ab7651/33ddaecbc11a8cef4eca0c21a9827f99"
```
Comment (vector-of-bool) [inline tools/get-win-openssl.ps1:15]: Was finally able to pin it down to a URL that appears to work correctly. Try this one:

```suggestion
$conan_ssl_path = "_/openssl/1.1.1h/_/8aee28bb7f844aa2e8e9f2b9e988027e/package/a79a557254fabcb77849dd623fed97c9c5ab7651/c200a0954463ed55a55d99809c0dba96"
```

**Review by vector-of-bool — CHANGES_REQUESTED** [review]

**Comment by vector-of-bool** [comment]
Sorry for the delay, the OpenSSL download thing was very confusion. I think it may be solved with the above change.

File: tools/get-win-openssl.ps1 (hunk 1 above)
Comment (Quincunx271) [reply tools/get-win-openssl.ps1]: Still doesn't appear to have worked. https://vector-of-bool.visualstudio.com/dds/_build/results?buildId=1526&view=logs&jobId=dc6ba0e6-a07f-5909-ef10-dfbfdafc5a6e

Comment (Quincunx271) [reply tools/get-win-openssl.ps1]: Feel free to make any commits you need to this PR to work out the issue; you don't need to wait for me to apply them if you don't want to.

Comment (vector-of-bool) [reply tools/get-win-openssl.ps1]: I'm losing my marbles on this one. I'll keep poking. Maybe I'll just vendor-in the OpenSSL tarball.

Comment (Quincunx271) [reply tools/get-win-openssl.ps1]: I went ahead and added a commit that embeds the OpenSSL tgz. The Windows test still fails, but it seems to be an error with permissions to remove a file.

**Comment by vector-of-bool** [comment]
Test timeout while compiling CryptoPP. Looks like the servers are running slow. I'm not very fond of that test case. May replace it in the future. I started the failed job again.

**Comment by vector-of-bool** [comment]
Thanks for going ahead and pulling in the OpenSSL tarball. Probably should have done that from the beginning. LGTM.

**Pull request merged**